src/applications/coronavirus-research/widgets/HeightWidget.jsx

```diff
@@ -36,61 +42,58 @@ export default class HeightWidget extends React.Component {
       if (this.isTouched(newState.touched)) {
         this.props.onBlur(this.props.id);
       }
+      this.labelStyleClasses = setLabelStyles(newState);
     });
   };
 
   handleChange = (field, value) => {
     let newState = _.set(['value', field], value, this.state);
     newState = _.set(['touched', field], true, newState);
-
     this.setState(newState, () => {
       if (this.props.required && this.isIncomplete(newState.value)) {
         this.props.onChange();
       } else {
         this.props.onChange(calculateHeight(newState.value));
       }
```

Comment:
```suggestion
      const height =
        this.props.required && this.isIncomplete(newState.value)
          ? calculateHeight(newState.value)
          : undefined;
      this.props.onChange(height);
      this.labelStyleClasses = setLabelStyles(newState);
```
does this work?

Reply:
Ternary is better way for sure, just had to reverse it:
`const height =
        this.props.required && this.isIncomplete(newState.value)
          ? undefined
          : calculateHeight(newState.value);
      this.props.onChange(height);
      this.labelStyleClasses = setLabelStyles(newState);`